lib/app/features/main/pages/home/todo_detail_screen.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_application_example/app/features/main/data/entity/todo_entity.dart';
+import 'package:flutter_application_example/app/features/main/data/repository/todo_repository.dart';
+import 'package:flutter_application_example/app/features/main/pages/home/update_todo_screen.dart';
+import 'package:flutter_application_example/app/widgets/status/loading_widget.dart';
+import 'package:flutter_application_example/data/api/todo/todo_update.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:riverpod_hook_mutation/riverpod_hook_mutation.dart';
+
+part 'todo_detail_screen.g.dart';
+
+@riverpod
+class Todo extends _$Todo {
+  TodoRepository get _repository => ref.read(todoRepositoryProvider);
+  @override
+  Future<TodoEntity> build(int id) {
+    return _repository.getDetail(id);
+  }
+
+  Future<bool> toggleStatus(int id) async {
+    final response = await _repository.toggle(id);
+    ref.invalidateSelf();
+    await future;
+    return response;
+  }
+
+  Future<TodoEntity> updateTodo(
+    int id,
+    TodoUpdate todo,
+  ) async {
+    final response = await ref.read(todoRepositoryProvider).update(
+          id,
+          todo,
+        );
+    ref.invalidateSelf();
+    await future;
+    return response;
+  }
+}
+
+class TodoDetailScreen extends HookConsumerWidget {
```

Comment:
Rename this to `TodoDetailPage` to follow the app naming convention
```suggestion
class TodoDetailScreen extends HookConsumerWidget {
```